classes/class.ilMumieTaskUserListGUI.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+/**
+ * MumieTask plugin
+ *
+ * @copyright   2019 integral-learning GmbH (https://www.integral-learning.de/)
+ * @author      Tobias Goltz (tobias.goltz@integral-learning.de)
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+/**
+ * This form is used to add, edit and validate MUMIE Server configurations
+ *
+ */
+class ilMumieTaskUserListGUI extends ilTable2GUI
+{
+    private $members;
+    private $parent_gui;
+    private $postvar;
+
+    public function __construct($parentObj, $form)
+    {
+        global $ilDB, $lng;
+        $this->setId("user" . $_GET["ref_id"]);
+        parent::__construct($parentObj, 'displayUserList');
+
+        $this->setFormName('participants');
+        $this->addColumn($lng->txt('rep_robj_xmum_frm_list_name'), 'name');
+        $this->addColumn($lng->txt('rep_robj_xmum_frm_list_grade'), 'note');
+        $this->addColumn($lng->txt('rep_robj_xmum_frm_list_submissions'), 'submission');
+        $this->setDefaultFilterVisiblity(true);
+
+        $members = $this->getMembers($parentObj, $form);
+
+        require_once('./Customizing/global/plugins/Services/Repository/RepositoryObject/MumieTask/classes/class.ilMumieTaskLPStatus.php');
+        ilMumieTaskLPStatus::updateGrades($parentObj->object);
+
+        $this->tpl->addBlockFile(
+            "TBL_CONTENT",
+            "tbl_content",
+            "tpl.mumie_user_list.html",
+            "Customizing/global/plugins/Services/Repository/RepositoryObject/MumieTask"
+        );
+
+        foreach ($members as $user_id) {
+            $this->tpl->setCurrentBlock("tbl_content");
+            $this->css_row = ($this->css_row != "tblrow1")
+            ? "tblrow1"
+            : "tblrow2";
+            $this->tpl->setVariable("CSS_ROW", $this->css_row);
+            $this->ctrl->setParameterByClass('ilObjMumieTaskGUI', 'user_id', $user_id);
+            $this->tpl->setVariable('LINK_NAME', $this->ctrl->getLinkTarget($parentObj, 'displayGradeList'));
+            $this->tpl->setVariable('LINK_TXT', $lng->txt('rep_robj_xmum_frm_list_change_grade'));
+
+            $result = $ilDB->query(
+                "SELECT mark
+                FROM ut_lp_marks 
+                WHERE usr_id = " . $ilDB->quote($user_id, "integer") .
+                " AND " .
+                "obj_id = " . $ilDB->quote($parentObj->object->getId(), "integer")
+            );
+
+            $grade = $ilDB->fetchAssoc($result);
```

Comment:
this should be extracted into a function that returns the grade as a number

Reply:
done